src/components/magic/adeptPowers/PowerSelectorComponent.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import 'styles/magic/SpellSelector.sass';
+import FilterTable from '../../FilterableTable';
+import powerData from '../../../data/powers.json';
+import PropTypeChecking from '../../../config/propTypeChecking';
+import PowerSelectedDisplay from './PowerSelectedDisplayComponent';
+import PowerCategoryHeader from './PowerCategoryHeader';
+import PowerDetailRow from './PowerDetailRow';
+import Modal from '../../ModalButtonComponent';
+
+class PowerSelectorComponent extends React.Component {
+	render() {
+		const {selectedPowers, pointsSpent, maxPoints, isMystic, karmaSpent, actions} = this.props;
+
+		// generated power details to populate addPowerModals
+		const powerRows = powerData.map((power, index) => {
+			const props = {isMystic, power, index, pointsSpent, add: true, maxPoints, actions};
+			return <PowerDetailRow {...props} key={power + index} />;
+		});
+
+		return (
+			<div className="powers">
+				<p>
+					Power Points: <strong>{pointsSpent}</strong>
+					{isMystic ?
+						<span> Karma Spent: <strong>{karmaSpent}</strong></span>
+						: null}
+				</p>
+				<div className="power-selector">
+					<div className="btn-group">
+						<Modal
+							key={'powers'}
+							modalName="Powers"
+							modalContent={
+								<div className="col">
+									<FilterTable tableData={{header: <PowerCategoryHeader />, body: powerRows}} />
```

Comment:
You might want to abstract out the FilterTable into a smart component that is connected to the redux store. You'll need to take the powerRows mapping with it, to generate it every time the redux store get's updated. Otherwise the modal is disconnected from its parent and will never update.

If you need an example; the ArmorModComponent, WeaponModComponent, and the SkillgroupsComponent are all modals that I had to tie into the redux store to get them to live update.

Reply:
Yeah, it also works to load the pointsSpent from state in the Row, this will cause the table to update properly.  I had worked on this and will put it in with the res tof the updates.

Reply:
If I do it my suggested way it makes testing not happy.  The problem is actually the modal not reconizing it needs to re-render.